src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypeLikelihoodCalculatorDRAGEN.java

```diff
@@ -0,0 +1,495 @@
+        final double[] outputArray = new double[genotypeCount];
+        Arrays.fill(outputArray, Double.NEGATIVE_INFINITY);
+
+        final Allele refAllele = sampleLikelihoods.getAllele(0);
+
+        //Determine the size of an allele page for the readsLikelihoodsByAlleleFrequency table
+        final int readCount = sampleLikelihoods.evidenceCount();
+        final int alleleDataSize = readCount * (ploidy + 1);
+
+        for(int gtAlleleIndex = 0; gtAlleleIndex < sampleLikelihoods.numberOfAlleles(); gtAlleleIndex++) {
+            //This is crufty, it just so happens that the index of the homozygous genotype corresponds to the maximum genotype count per field.
+            //This should be pulled off as a calculator in some genotyping class.
+            final int indexForGT = calculators.genotypeCount(ploidy, gtAlleleIndex + 1) - 1;
+            double[] readLikelihoodsForGT = readLikelihoodsByGenotypeIndex[indexForGT];
+
+            for(int errorAlleleIndex = 0; errorAlleleIndex < sampleLikelihoods.numberOfAlleles(); errorAlleleIndex++) {
+                // We only want to make calls on SNPs for now
+                if (sampleLikelihoods.getAllele(gtAlleleIndex) == sampleLikelihoods.getAllele(errorAlleleIndex) ||
+                        sampleLikelihoods.getAllele(gtAlleleIndex).length() != refAllele.length() ||
+                        sampleLikelihoods.getAllele(errorAlleleIndex).length() != refAllele.length()) {
+                    continue;
+                }
+                // TODO super validate this
+                byte baseOfErrorAllele = sampleLikelihoods.getAllele(errorAlleleIndex).getBases()[0];
+
+                double forwardHomopolymerAdjustment = FRDBQDUtils.computeForwardHomopolymerAdjustment(paddedReference, offsetForRefIntoEvent, baseOfErrorAllele);
+                double reverseHomopolymerAdjustment = FRDBQDUtils.computeReverseHomopolymerAdjustment(paddedReference, offsetForRefIntoEvent, baseOfErrorAllele);
+
+                // This selects the index for the reads page in the table corresponding to the likelihoods of the read given allele frequency of 1
+                final int offsetForReadLikelihoodGivenAlleleIndex = alleleDataSize * errorAlleleIndex + readCount;
+
+                // BQD scores by strand
+                double minScoreFoundForwardsStrand = computeBQDModelForStrandData(strandForward, forwardHomopolymerAdjustment, readLikelihoodsForGT, offsetForReadLikelihoodGivenAlleleIndex, true, errorAlleleIndex);
+                double minScoreFoundReverseStrand = computeBQDModelForStrandData(strandReverse, reverseHomopolymerAdjustment, readLikelihoodsForGT, offsetForReadLikelihoodGivenAlleleIndex, false, errorAlleleIndex);
+
+                double modelScoreInLog10 = (minScoreFoundForwardsStrand + minScoreFoundReverseStrand)/-10.0;
```

Comment:
Which method? In GATK aren't we in log10 rather than natural log and consequently that is the correct conversion? The existing method seems to use natural logs and there are many permutations in there.